tests/pos/i2051.scala

```diff
@@ -0,0 +1,2 @@
+class A[T](val x:T)
```

Comment:
it would be nice to add several fields to the body of the class that are overriden in a subclass.

It would also be nice to include tests with trait vals and trait constructors.